### ISSUE TITLE
Support for vmagent cluster setup

### ISSUE THREAD

**Issue by alexandrst88** [VictoriaMetrics/operator#177]
Ref doc: https://victoriametrics.github.io/vmagent.html#scraping-big-number-of-targets

It requires to identify each member in cluster,

```
/path/to/vmagent -promscrape.cluster.membersCount=2 -promscrape.cluster.memberNum=0
/path/to/vmagent -promscrape.cluster.membersCount=2 -promscrape.cluster.memberNum=1 
```
**vmagent** is deployed via Kubernetes Deployment, I assume it's impossible to identify correctly each member in case of Rolling  Upgrade and etc. It could be achievable via Kubernetes StatefulSet and [downward API ](https://kubernetes.io/docs/tasks/inject-data-application/downward-api-volume-expose-pod-information/)

**Comment by f41gh7** [comment]
In addition, statefullset implementation will allow to use pvc claim and do not lose persistent queue state after restart.

 index can be obtained with `


```
  - name: promscrape.cluster.memberNum
    valueFrom:
      fieldRef:
        fieldPath: metadata.annotations['spec.pod.beta.kubernetes.io/statefulset-index']
```
 and set membersCount with arg according to the number of replicas.

 Note, this solution may lead to metrics loss on statefulset upgrade.

**Comment by TinySong** [comment]
> Note, this solution may lead to metrics loss on statefulset upgrade.

setting .spec.template.updateStrategy.type to OnDelete will Alleviate this problem. Let users decide when to update.

**Comment by TinySong** [comment]
any progress? if not, I will finish this feature.

**Comment by f41gh7** [comment]
> any progress? if not, I will finish this feature.

going to add it in next few days. I've made few experiments with design.

**Comment by f41gh7** [comment]
Added to master branch, example usage, produces 3 shards with 2 replicas per shard:
```yaml
apiVersion: operator.victoriametrics.com/v1beta1
kind: VMAgent
metadata:
  name: example-vmagent
spec:
  serviceScrapeNamespaceSelector: {}
  podScrapeNamespaceSelector: {}
  podScrapeSelector: {}
  serviceScrapeSelector: {}
  scrapeInterval: 60s
  vmAgentExternalLabelName: vmagent-ha
  shardCount: 3
  replicaCount: 2
  remoteWrite:
    - url: "http://vmsingle-example-vmsingle-persisted.default.svc:8429/api/v1/write"
```

**Comment by TinySong** [comment]
Great, thank you for your wonderful work !

**Comment by f41gh7** [comment]
Included to 0.12.0 release
